php/Blockonomics.php

```diff
@@ -217,9 +227,21 @@ private function set_headers($api_key)
         }
     }
 
+
     public function testSetup()
     {
-        $response = $this->get_callbacks();
+        $api_key = get_option("blockonomics_api_key");
+        $BCH_Enabled = get_option('blockonomics_bch');
+        if ((!isset($api_key) || strlen($api_key) != 43) && isset($BCH_Enabled)){
```

Comment:
`isset($BCH_Enabled)` returns true if this setting has been used.

![BCH test setup](https://user-images.githubusercontent.com/39666372/105808976-7adbb000-5fb1-11eb-8c36-153c10ad0623.png)

`if(!$api_key && $BCH_Enabled)`  is sufficient

Reply:
changed it.